File: 20 Valid Parentheses/isValid.cpp
class Solution {
public:
    bool isValid(string s) {
        
        stack<char> stack;     
        for( int i = 0; i < s.size(); i++){            
            if( s[i] == '(' || s[i] == '[' || s[i] == '{')
                stack.push( s[i] );           
            else{
                if( stack.size() == 0 )
                    return false;
                
                char c = stack.top();
                stack.pop();
                
                char match;
                switch( s[i] ){
                    case ')': match = '('; break;
                    case ']': match = '['; break;
                    case '}': match = '{'; break;
                }
                
                if( c != match )
                    return false;
            }
        }
        
        if( stack.size() != 0 )
            return false;

        return true;
    }
};

File: 20 Valid Parentheses/isValid.java
public class Solution {
    public boolean isValid(String s) {
        
        Stack<Character> stack = new Stack();
        
        for(int i = 0; i < s.length(); i++){
            if( s.charAt(i) == '(' || s.charAt(i) == '[' || s.charAt(i) == '{')
                stack.push( s.charAt(i) );
            else{
                if( stack.empty() ) // stack.size() == 0 also works
                    return false;
                
                char c = stack.peek();
                stack.pop();
                
                char match = ' ';
                switch( s.charAt(i) ){
                        case')': match = '('; break;
                        case']': match = '['; break;
                        case'}': match = '{'; break;
                }
                
                if( c != match )
                    return false;
            }
        }
        
        if( !stack.empty() )
            return false;
        
        return true;
    }
}